src/Umbraco.Infrastructure/Persistence/Repositories/Implement/KeyValueRepository.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NPoco;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Migrations;
+using Umbraco.Core.Models;
+using Umbraco.Core.Persistence.Dtos;
+using Umbraco.Core.Persistence.Querying;
+using Umbraco.Core.Scoping;
+using Umbraco.Infrastructure.Persistence.Repositories;
+
+namespace Umbraco.Core.Persistence.Repositories.Implement
+{
+    internal class KeyValueRepository : NPocoRepositoryBase<string, IKeyValue>, IKeyValueRepository
```

Comment:
See other comment about repository design, if we are consistent with the `IReadRepository` and `IWriteRepository`, etc... then the repository methods are already there to deal with. Having GetValue, SetValue aren't consistent with what we have.

Reply:
Have removed these now, they are still in the service layer and implemented using methods that act on the `IKeyValue/KeyValue` entity class now.